### PR TITLE
remove slice2java

### PR DESCRIPTION
**Pull request by jburel**
Remove slice2java.
Check that the build is green

### CODE REVIEW

**Review by sbesson — CHANGES_REQUESTED** [review]
Although the build passes without issue with this PR included, both in the GitHub Actions context as well as locally (without slice2java installed), the server failed to start - see https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-server/1234/

Looking quickly at the logs, the services did not get started. Looking at a few configuration files, il looks like a few variables did not get subsituted e.g. in `etc/omero.properties`:

```
omero.version=5.6.3-242-d748290-ice${versions.ice_lib}-b1224
```

**Comment by jburel** [comment]
Since we are using only one ice version, that could be removed.

**Comment by jburel** [comment]
Back to green https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-server/1242/ with this PR included

**Comment by sbesson** [comment]
OMERO.script integration tests (Python + Java) have been failing for the last two days - see https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-test-integration/1137/. This is possibly related to these changes

**Comment by jburel** [comment]
https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-test-integration/1144/ is green with this PR included

**Comment by sbesson** [comment]
In general, :+1: for getting rid of the unnecessary `slice2java` dependency for building. 

The biggest change looking at the nightly artifacts is the removal of the `-ice36` suffix from `omero.version` and hence the OMERO.server artifact name.

This raises essentially the same question as in https://github.com/ome/openmicroscopy/pull/6318#issuecomment-1113366461. Do we have an authoritative list of all the downstream locations that would be impacted by the change and will need to be reviewed? And do you have a feeling of the cost/time of updating these post server release?

**Comment by joshmoore** [comment]
Briefly:
 * I don't have an idea of what impact dropping `-ice36` might have.
 * Is slice2py still needed?

**Comment by jburel** [comment]
* omego and the ansible roles are the ones to review. 
I have rolled back the name change. This does not need to be included in that body of work.

**Comment by jburel** [comment]
@joshmoore I think we can also drop ``slice2py`` to build 
I have run ``./build.py clean && ./build.py`` in a conda environment without slice2py and it works fine.
The question is: do we want to have env to build only and env to run only?

**Comment by jburel** [comment]
The last 2 commits remove:
*  slice2py 
* ice-action

the build remains green

File: components/antlib/scripts/parse_version [hunk 1]
```diff
@@ -82,7 +82,7 @@ def snapshot_logic(omero_version):
     if m:
         # Non-tag version
         base = incr_version(m.group("BASE"))
-        omero_version = "%s%s-SNAPSHOT" % (
```
Comment (sbesson) [inline components/antlib/scripts/parse_version]: Should this change be reverted if the `-ice` separator is maintained?

Comment (jburel) [reply components/antlib/scripts/parse_version]: well spotted

**Review by sbesson — APPROVED** [review]
Thanks, the latest version of this PR feels like the minimal set of changes allowing to drop the unnecessary Ice requirement at the level of this repository without modifying artifact names and downstream consumptions.

No objection to merging from my side

**Review by joshmoore — APPROVED** [review]
Also no objections. I imagine eventually, we strip out the rest of strip2cpp, etc. but happy to wait until we drop ant & ivy :smile:

**Pull request merged**